var/spack/repos/builtin/packages/py-smartsim/package.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+
+from spack.package import *
+
+
+class PySmartsim(PythonPackage):
+    """A workflow library to easily integrate machine learning libraries with
+    high performance computing simulations and applications
+    """
+
+    homepage = "https://www.craylabs.org/docs/overview.html"
+    git = "https://github.com/CrayLabs/SmartSim"
+    pypi = "smartsim/smartsim-0.4.2.tar.gz"
+
+    maintainers("MattToast")
+
+    version("0.5.0", sha256="35b36243dc84af62261a7f772bae92f0b3502faf01401423899cb2a48339858c")
+    version("0.4.2", sha256="ab632ff7d036e73822ddc5081fe85ea69c48d8be53ad0e6e487e9193eb3410f6")
+
+    variant("torch", default=True, description="Build with the pytorch backend")
+    variant("cuda", default=False, description="Use CUDA")
+    variant("rocm", default=False, description="Use ROCm")
+
+    depends_on("python@3.8:3.10", type=("build", "run"))
```

Comment:
Where do you see this upper bound?

Reply:
Whoops! Force of habit. SmartSim when built traditionally through its home-brewed CLI is built against PyTorch 1.11 which is only supported through Py 3.10, so in the github CI for instance, you will see that it is mainly tested against Py 3.8-3.10.

However there is no reason for the limit if RedisAI is built against newer version of torch. Will remove!

Reply:
Still need to remove upper bound. Alternatively, we could stick to only supporting the versions of Python for which wheels exist on PyPI. I've done that in other packages.

Reply:
You had actually correctly pointed out in another piece of feedback that this is strictly enforced in the `setup.cfg`. The upper bound should stay.